Eric s geo coder

ESmith92 opened the pull request:
update url builder
    -rename google_builder() to reverse_google_builder()
    -add forward_google_builder()
    -add address variable
    -add set_address()
    -update function name in test and google-revrse-geocoder class

update google-reverse-geocoder
    -rename get_url() to get_reverse_url()
    -add get_forward_url()
    -move service call and json decode to function get_data()
    -update get_locations() to use get_data()
    -add get_coordinates() to get coords from a "address"
    -update spec test to use get_reverse_url()
    -update implementation to force new functions()

create parser for the forward geo coding
    -add class to loop through the json from google forward geo coding
        -loops through results and selects the coordinates
        -checks if coordinate is already in the array
        -returns an array of GeoCoordinates
    -update google-reverse-geocoder to use the forward parser for the forward geo coding

create test for the forward geo coding
    -create unit testing to check that forward geo parser is initalizable
    -create spec tests for the forward geo coding
        -update googleReverseGeocoderSpec to have test for the forward geo coding
            -add test to check url builder returned url
            -add test to run through the process of forward geo coding
        -create spec tests for the forward geo coding parser
    -create integration testing with google's forward geo coding service

ESmith92 committed:
update url builder to work with forward geo coder

	-rename google_builder() to reverse_google_builder()
	-add forward_google_builder()
	-add address variable
	-add set_address()
	-update function name in test and google-revrse-geocoder class

ESmith92 committed:
update google-reverse-geocoder

	-rename get_url() to get_reverse_url()
	-add get_forward_url()
	-move service call and json decode to function get_data()
	-update get_locations() to use get_data()
	-add get_coordinates() to get coords from a "address"
	-update spec test to use get_reverse_url()
	-update implementation to force new functions()

ESmith92 committed:
remove white space issues

ESmith92 committed:
create parser for the forward geo coding

	-add class to loop through the json from google forward geo coding
		-loops through results and selects the coordinates
		-checks if coordinate is already in the array
		-returns an array of GeoCoordinates
	-update google-reverse-geocoder to use the forward parser for the
forward geo coding

ESmith92 committed:
create test for the forward geo coding

	-create unit testing to check that forward geo parser is initalizable
	-create spec tests for the forward geo coding
		-update googleReverseGeocoderSpec to have test for the forward geo
coding
			-add test to check url builder returned url
			-add test to run through the process of forward geo coding
		-create spec tests for the forward geo coding parser
	-create integration testing with google's forward geo coding service

ESmith92 committed:
fix whitespace issues

ESmith92 committed:
Merge branch 'develop' into EricS-GeoCoder

rraub commented on src/providers/google-reverse-geocoder.php:
```diff
@@ -29,7 +29,8 @@ class GoogleReverseGeocoder implements ReverseGeocoder {
     Support\JsonDecoder $geo_json_decoder = null,
-    Support\GoogleReverseGeocoderParser $geo_parser = null ) {
+    Support\GoogleReverseGeocoderParser $geo_parser = null,
+    Support\GoogleForwardGeocoderParser $geo_parser2 = null ) {
```
geo_parser2.. really.. shorter variable names are not faster, they are just more confusing. 😕

rraub commented:
Sorry it took me so long to look at this!

ESmith92 committed:
Update google-reverse-geocoder.php

-change geo_parser and geo_parser to reverse_geo_parser and forward_geo_parser

ESmith92 committed:
update google-reverse-geocoder

	-update constructor if statements to use new parser names
		-$reverse_geo_parser
		-$forward_geo_parser

ESmith92 committed:
Merge branch 'develop' into EricS-GeoCoder

ESmith92 commented:
It broke while composer was updating and I don't know how to rerun the test without just making a new pull request

rraub commented:
I restarted the tests for you, thanks for making the corrections!
